e3sm_diags/derivations/utils.py

```diff
@@ -0,0 +1,309 @@
+        var.attrs["units"] = target_units
+    elif var.name == "AOD_550_ann":
+        var.attrs["units"] = target_units
+    elif var.name == "AOD_550":
+        var.attrs["units"] = target_units
+    elif var.attrs["units"] == "C" and target_units == "DegC":
+        var.attrs["units"] = target_units
+    elif var.attrs["units"] == "N/m2" and target_units == "N/m^2":
+        var.attrs["units"] = target_units
+    elif var.name == "AODVIS" or var.name == "AOD_550_ann" or var.name == "TOTEXTTAU":
+        var.attrs["units"] = target_units
+    elif var.attrs["units"] == "fraction":
+        var = 100.0 * var
+        var.attrs["units"] = target_units
+    elif var.attrs["units"] == "mb":
+        var.attrs["units"] = target_units
+    elif var.attrs["units"] == "gpm":  # geopotential meter
+        var = var / 9.8 / 100  # convert to hecto meter
+        var.attrs["units"] = target_units
+    elif var.attrs["units"] == "Pa/s":
+        var = var / 100.0 * 24 * 3600
+        var.attrs["units"] = target_units
+    elif var.attrs["units"] == "mb/day":
+        var = var
+        var.attrs["units"] = target_units
+    elif var.name == "prw" and var.attrs["units"] == "cm":
+        var = var * 10.0  # convert from 'cm' to 'kg/m2' or 'mm'
+        var.attrs["units"] = target_units
+    elif var.attrs["units"] in ["gC/m^2/s"] and target_units == "g*/m^2/day":
+        var = var * 24 * 3600
+        var.attrs["units"] = var.attrs["units"][0:7] + "day"
+    elif (
+        var.attrs["units"] in ["gN/m^2/s", "gP/m^2/s"] and target_units == "mg*/m^2/day"
+    ):
+        var = var * 24 * 3600 * 1000.0
+        var.attrs["units"] = "m" + var.attrs["units"][0:7] + "day"
+    elif var.attrs["units"] in ["gN/m^2/day", "gP/m^2/day", "gC/m^2/day"]:
+        pass
+    else:
+        temp = udunits(1.0, var.attrs["units"])
+        coeff, offset = temp.how(target_units)
+
+        # Keep all of the attributes except the units.
+        with xr.set_options(keep_attrs=True):
+            var = coeff * var + offset
+
+        var.attrs["units"] = target_units
+
+    return var
+
+
+def _apply_land_sea_mask(
+    var: xr.DataArray, var_mask: xr.DataArray, lower_limit: float
+) -> xr.DataArray:
+    """Apply a land or sea mask on the variable.
+
+    Parameters
+    ----------
+    var : xr.DataArray
+        The variable.
+    var_mask : xr.DataArray
+        The variable mask ("LANDFRAC" or "OCNFRAC").
+    lower_limit : float
+        Update the mask variable with a lower limit. All values below the
+        lower limit will be masked.
+
+    Returns
+    -------
+    xr.DataArray
+        The masked variable.
+    """
+    cond = var_mask > lower_limit
+    masked_var = var.where(cond=cond, drop=False)
+
+    return masked_var
+
+
+def adjust_prs_val_units(
+    prs: "FileAxis", prs_val: float, prs_val0: Optional[float]
+) -> float:
+    """Adjust the prs_val units based on the prs.id"""
+    # FIXME: Refactor this function to operate on xr.Dataset/xr.DataArray.
```

Comment:
Is looks like this still has to be done?

Reply:
Not yet, it needs to be refactored in #716.

Reply:
I'll move all of these #716 related items over to this issue.